### PR TITLE
Refactored the similar events

### PR DESCRIPTION
**Pull request by nikomakela**
HH-83. The similar events got a new refinement: The similar events should be searched with the current event's age group and categories. Also the rendering is now done with the components from the HCRC.

image

### CODE REVIEW

File: src/domain/event/queryUtils.ts [hunk 1]
```diff
@@ -1,52 +1,93 @@
...
+ * Filter the similar events query results:
+ * - Show unique results only
+ * - Don't show sibling events / events from the same super event
+ *
+ * NOTE: LinkedEvents-query with keyword_OR_set* -param returns can return as many duplicates
+ * as there are keywords listed.
+ *
+ * TODO: This could be simplified if the LinkedEvents -query would be fixed and it would return distinct results only.
+ */
+const _filterSimilarEvents = (event: EventFields, data: EventFields[]) =>
```
Comment (nikomakela) [inline src/domain/event/queryUtils.ts:57]: NOTE: there is still an issue with similar events query, and the workaround makes a mess in the code.

- [ ] Clear the bubblegum workaround when the LinkedEvents has fixed their keyword-OR-query.

**Comment by nikomakela** [comment]
This PR needs this to be merged also https://github.com/City-of-Helsinki/react-helsinki-headless-cms/pull/51

File: src/common-events/utils/headless-cms/headlessCmsUtils.tsx [hunk 2]
```diff
@@ -141,6 +141,8 @@ export const getDefaultCollections = (
         description: collection.description,
         type: getCollectionUIType(collection),
         collectionContainerProps: { withDots: true },
+        // TODO: The collection specific URL from the module itself should be used instead of a hard coded value.
+        showAllUrl: '/',
```
Comment (melniiv) [inline src/common-events/utils/headless-cms/headlessCmsUtils.tsx:145]: so do we need the actual value right now, or we use empty instead and hiding the link for right now?

**Review by melniiv — APPROVED** [review]
one small comment, otherwise looks good

File: src/common-events/utils/headless-cms/headlessCmsUtils.tsx (hunk 2 above)
Comment (nikomakela) [reply src/common-events/utils/headless-cms/headlessCmsUtils.tsx:145]: when the url is not given, no link is shown. The link should come from the CMS inside the module attributes. It's ordered, but not implemented properly yet -- as far as I know

File: src/domain/event/queryUtils.ts (hunk 1 above)
Comment (nikomakela) [reply src/domain/event/queryUtils.ts:57]: The LinkedEvents team said that the results should now be done with distinct.

**Pull request merged**